### ISSUE TITLE
message showing failed to send state, but unable to resend

### ISSUE THREAD

**Issue by AndrewSB** [GetStream/stream-chat-swiftui#548]
## What did you do?

GET `https://chat.stream-io-api.com/messages/0c9b8de3-730d-4da9-9f17-ad428223af81?api_key=xxx` looks like 
```
{"message":{"updated_at":"2024-04-30T21:07:26.730429Z","reply_count":0,"channel":{"id":"!members-FQ9qla_bKvBJriAjp1jRo6cNcot3nzk14PK8lsaObdU","show_to_user_61da9ab70dd698752ad4a0a4":true,"created_at":"2022-05-05T05:54:15.984854Z","created_by":{"id":"5d39929d0febe07fc814e06b","banned":false,"last_active":"2024-07-15T20:56:17.907861805Z","created_at":"2022-03-24T16:37:46.123555Z","image":"https://storage.googleapis.com/production-avatars/ofiCKFaShjPWQEEuryDvQYDiI","updated_at":"2024-07-15T18:48:06.140259Z","username":"fru","role":"user","date_of_birth":"1988-10-08T00:00:00","online":true,"signup_date":"2019-07-25T11:29:33.026000","name":"Josh Fruhman"},"disabled":false,"type":"messaging","cid":"messaging:!members-FQ9qla_bKvBJriAjp1jRo6cNcot3nzk14PK8lsaObdU","frozen":false,"member_count":2,"updated_at":"2023-09-18T15:29:36.403921Z","config":{"typing_events":true,"reminders":false,"connect_events":true,"reactions":false,"replies":true,"uploads":true,"push_notifications":true,"updated_at":"2024-03-22T19:58:18.667392Z","mark_messages_pending":false,"polls":false,"automod_behavior":"flag","url_enrichment":true,"commands":[],"mutes":true,"custom_events":true,"message_retention":"infinite","name":"messaging","automod":"disabled","quotes":true,"automod_thresholds":{},"created_at":"2022-04-28T18:14:10.661334Z","max_message_length":5000,"blocklist_behavior":"flag","read_events":true,"search":true},"show_to_user_5d39929d0febe07fc814e06b":true,"last_message_at":"2024-07-15T21:08:24.775448Z"},"html":"<p>i figured that’s the case</p>\n","deleted_reply_count":0,"type":"regular","own_reactions":[],"cid":"messaging:!members-FQ9qla_bKvBJriAjp1jRo6cNcot3nzk14PK8lsaObdU","id":"0c9b8de3-730d-4da9-9f17-ad428223af81","user":{"id":"61da9ab70dd698752ad4a0a4","banned":false,"signup_date":"2022-01-09T08:20:07.392000","blocked_user_ids":[],"last_active":"2024-07-15T21:09:38.21723889Z","created_at":"2022-04-01T06:55:40.652883Z","image":"https://storage.googleapis.com/production-avatars/wSQiZEQNQkCWJSMickTnTzhMK","language":"","updated_at":"2024-07-15T21:09:37.043533Z","role":"user","date_of_birth":"1995-04-20T07:20:07","username":"emirhanerkan","teams":[],"online":true,"name":"Emir"},"shadowed":false,"pinned_at":null,"pin_expires":null,"attachments":[],"reaction_counts":{},"reaction_scores":{},"silent":false,"text":"i figured that’s the case ","pinned_by":null,"created_at":"2024-04-30T21:07:26.730429Z","mentioned_users":[],"pinned":false,"latest_reactions":[]},"duration":"7.61ms"}
```

which renders this way
![Bug 1160 Reclip beta](https://github.com/user-attachments/assets/8286356e-432b-48f7-85f0-64827db49c3e)

tapping the red icon shows options to delete + resend

## What did you expect to happen?

tapping resend would resolve the unsent message, and get rid of the red icon

## What happened instead?

no change, looking at network logs i see `POST https://chat.stream-io-api.com/channels/messaging/!members-FQ9qla_bKvBJriAjp1jRo6cNcot3nzk14PK8lsaObdU/message?api_key=xxx`

<img width="985" alt="Screenshot 2024-07-15 at 2 16 55 PM" src="https://github.com/user-attachments/assets/148a1eb0-635a-4bc4-a9b7-137de4f89cff">

## GetStream Environment
**GetStream Chat version:** 4.52

**Comment by martinmitrevski** [comment]
Hi @AndrewSB,

Are you sure about the StreamChat version? In the last screenshot, it seems like the version 4.48 is used. This issue was fixed in our 4.50 release: https://github.com/GetStream/stream-chat-swift/pull/3061.

Hope that helps, Martin

**Comment by AndrewSB** [comment]
my mistake, this came from a user on an outdated client. i'll have them update & check back

as an aide- i'd love to use your built xcframework and get rid of my fork. if you could do a major release anytime soon, please include https://github.com/GetStream/stream-chat-swiftui/pull/430
